pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -4280,7 +4435,7 @@ def PumpCostRule(b, l1, l2):
                         * mh.p_rhog
                         * 1e3  # convert the kUSD/kWh to kUSD/Ws
                         * sum(
-                            b.v_PumpHead[l1, l2, t]
+                            1609
```

Comment:
Please consider exploring alternative methods to linearize the pump cost function in the co-optimize approach rather than replacing the pump head with constant values. Co-optimize is designed to incorporate pump cost within the objective function, accounting for elevation changes and frictional losses. Using a constant pump head overlooks the critical impact of elevation changes and frictional losses in the network on the objective cost. Without variable pump heads, hydraulic post-processing might suffice instead of co-optimize.

Additionally, it's important to note that we cannot use a constant number from a small case study and generalize it to all other case studies.

Reply:
Thanks  for the comment and feedback! The pricing for the variable component of pump now has the pump head variable. The linearization is now done based on the flow. Say grid size for flow is 10000 bbl/day. Then for 0<flow<10000, the flow is assumed to be constant 0, for flow 10000<flow<20000, the flow is assumed to be constant 10000, and so on.